portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,87 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+private List<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentJson = convertToJson(comments);
```

Comment:
You're so close to having the datastore piece complete!

I'd recommend taking a look at the "doGet" function in ListTasksServlet.java for how to get the comments out of your datastore now that you've done the work putting them in there during "doPost".

After that you can get remove your `private List<string> comments;` class variable because you'll pull the comments directly from DataStore on ever GET request.